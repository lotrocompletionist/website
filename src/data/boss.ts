export interface IBoss {
  id: number;
  name: string;
}
