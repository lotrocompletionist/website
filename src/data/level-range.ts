export interface ILevelRange {
  minimum: number;
  maximum?: number;
}
