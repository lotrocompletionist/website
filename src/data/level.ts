export interface Level {
  minimum: number;
  maximum?: number;
}
