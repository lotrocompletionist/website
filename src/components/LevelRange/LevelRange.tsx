import * as React from "react";
import { ILevelRange } from "../../data/level-range";

export interface LevelRangeProps {
  range: ILevelRange;
}

export function LevelRange(props: LevelRangeProps) {
  const maximum = props.range.maximum ? " - " + props.range.maximum : "";

  return (
    <>
      Level: {props.range.minimum}
      {maximum}
    </>
  );
}
