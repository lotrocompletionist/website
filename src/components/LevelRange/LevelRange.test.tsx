import * as React from "react";
import { create } from "react-test-renderer";
import { LevelRange } from "./LevelRange";
import { ILevelRange } from "../../data/level-range";

test("<LevelRange> renders correctly with minimum and maximum specified", () => {
  const levelRange: ILevelRange = {
    minimum: 10,
    maximum: 20
  };
  const tree = create(<LevelRange range={levelRange} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("<LevelRange> renders correctly with only mimimum specified", () => {
  const levelRange: ILevelRange = {
    minimum: 10
  };
  const tree = create(<LevelRange range={levelRange} />).toJSON();
  expect(tree).toMatchSnapshot();
});
