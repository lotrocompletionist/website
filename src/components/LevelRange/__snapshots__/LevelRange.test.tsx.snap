// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelRange> renders correctly with minimum and maximum specified 1`] = `
Array [
  "Level: ",
  "10",
  " - 20",
]
`;

exports[`<LevelRange> renders correctly with only mimimum specified 1`] = `
Array [
  "Level: ",
  "10",
  "",
]
`;
